lib/shopify_import_massager.ex

```diff
@@ -136,13 +227,46 @@ defmodule ShopifyImportMassager do
     File.close(output_file)
   end
 
-  def substitute_input_line(input_line, substitution_map) do
-    substitution_map
-    |> Enum.reduce(
-      input_line,
-      fn %{order_name: order_name, replace_order_number: replace_order_number}, adjusted_line ->
-        String.replace(adjusted_line, order_name, replace_order_number, global: true)
-      end
-    )
+  def get_column_index(file_type) do
+    # If it's an orders file, then only substitute on the first column.
+    # If it's a transaction, then only substitute on the second column.
+    # If it's a return, then only substitute on the first column.
+    # Remember: 0-based indexing
+    case file_type do
+      :orders -> 0
+      :transactions -> 1
+      :returns -> 0
+    end
+  end
+
+  # TODO It is slow and inefficient to do a line-by-line substitution for each
+  # input-output string, but sometimes orders names are referenced in a
+  # comment field. This is probably not an issue because we're trying to
+  # extract financial information such as sales values from the orders,
+  # not act as a source of truth.
+  #
+  # One approach is to do global substitutions across the whole file. That may
+  # or may not be faster.
+  #
+  # Substitutions are done only matched on the given column, but done across
+  # the whole line. This may or may not be wise.
+  def substitute_input_line(%{
+        input_line: input_line,
+        substitution_map: substitution_map,
+        column_index: column_index
+      }) do
+    match_column_value = input_line |> String.split(",") |> Enum.fetch!(column_index)
+    # IO.puts("match_column_value: #{inspect(match_column_value)}")
+
+    case Map.get(substitution_map, match_column_value, nil) do
+      nil ->
+        input_line
+
+      replace_order_number ->
+        # TODO Do we really want to globally replace on order_name on the whole line?
+        # IO.puts("Replacing #{match_column_value} with #{replace_order_number} in #{input_line}")
+        String.replace(input_line, match_column_value, replace_order_number, global: true)
+        # |> IO.inspect(label: "replacement line")
+    end
```

Comment:
Note that this module is quite long and is a candidate for refactoring into separate components.